substrate-node/pallets/pallet-minting/src/lib.rs

```diff
@@ -0,0 +1,468 @@
+        // Reset counters
+        node_report.counters = types::NodePeriodInformation::default();
+    }
+
+    // This method will calculate the minting rewards for all the periods that the node has accumulated
+    // Can only called by the farmer that owns the node
+    pub fn payout_node_periods(
+        account_id: &T::AccountId,
+        node_id: u32,
+    ) -> DispatchResultWithPostInfo {
+        let node = pallet_tfgrid::Nodes::<T>::get(node_id)
+            .ok_or(pallet_tfgrid::Error::<T>::NodeNotExists)?;
+
+        let twin_id = pallet_tfgrid::TwinIdByAccountID::<T>::get(&account_id)
+            .ok_or(pallet_tfgrid::Error::<T>::TwinNotExists)?;
+
+        let farm = pallet_tfgrid::Farms::<T>::get(node.farm_id)
+            .ok_or(pallet_tfgrid::Error::<T>::FarmNotExists)?;
+
+        ensure!(
+            twin_id == farm.twin_id,
+            Error::<T>::UnauthorizedToTriggerPayout
+        );
+
+        let mut payable_periods = PayablePeriods::<T>::get(node_id);
+        payable_periods = payable_periods
+            .into_iter()
+            .filter(|period| {
+                match Self::payout_period(&node, node.farm_id, node.connection_price, period) {
+                    Ok(_) => true,
+                    Err(e) => {
+                        log::debug!("payout failed: {:?}", e);
+                        false
+                    }
+                }
+            })
+            .collect();
+        PayablePeriods::<T>::insert(node_id, payable_periods);
+
+        Ok(().into())
+    }
+
+    pub fn payout_period(
+        node: &pallet_tfgrid::TfgridNode<T>,
+        farm_id: u32,
+        connection_price: u32,
+        period: &types::NodePeriodInformation,
+    ) -> DispatchResultWithPostInfo {
+        log::debug!("min capacity: {:?}", period.min_capacity);
+        let (cu, su) = period.min_capacity.cloud_units_permill();
+        log::debug!("cu: {}", cu);
+        log::debug!("su: {}", su);
+        let period_length = T::PeriodTreshold::get();
+
+        let farming_policy = pallet_tfgrid::FarmingPoliciesMap::<T>::get(period.farming_policy);
+        let farm = pallet_tfgrid::Farms::<T>::get(farm_id)
+            .ok_or(pallet_tfgrid::Error::<T>::FarmNotExists)?;
+        let farm_twin = pallet_tfgrid::Twins::<T>::get(farm.twin_id)
+            .ok_or(pallet_tfgrid::Error::<T>::TwinNotExists)?;
+
+        let mut cu_reward = 0;
+        let mut su_reward = 0;
+
+        let uptime_percentage = (period.uptime * 1_000 / period_length) as u128;
+
+        if farming_policy.id == 1 || farming_policy.id == 2 {
+            // CU/SU rewards are the amount of CU/SU for the minimal capacity scaled on a period
+            cu_reward = cu * farming_policy.cu as u64;
+            su_reward = su * farming_policy.su as u64;
+        } else {
+            let minimal_uptime_met =
+                period.uptime < (period_length * farming_policy.minimal_uptime as u64) / 1000;
```

Comment:
Also `uptime_percentage` could be reused here with something like:

`uptime_percentage < farming_policy.minimal_uptime`

Reply:
Uptime percentage is expressed as a percentage, in the minimal uptime met check we calculate how much uptime the farmer actually needs, It's different

Reply:
Yes, it's different because you compare time instead of percent
But it is mathematically the same

if you start with:
`period.uptime < (period_length * minimal_uptime) / 1000`

you multiply by 1000 each side
`period.uptime *1000 < period_length * minimal_uptime`

you divide each side by period_length
`period.uptime *1000 /  period_length < minimal_uptime`

which is the same as 
`uptime_percentage < minimal_uptime`